src/server/wesnothd/server.cpp

```diff
@@ -590,6 +606,53 @@ void server::dump_stats(const boost::system::error_code& ec)
 	start_dump_stats();
 }
 
+void server::start_dummy_player_updates()
+{
+#if BOOST_VERSION >= 106600
+	dummy_player_timer_.expires_after(std::chrono::seconds(30));
+#else
+	dummy_player_timer_.expires_from_now(std::chrono::seconds(30));
+#endif
+	dummy_player_timer_.async_wait([this](const boost::system::error_code& ec) { dummy_player_updates(ec); });
+}
+
+void server::dummy_player_updates(const boost::system::error_code& ec)
+{
+	if(ec) {
+		ERR_SERVER << "Error waiting for dummy player timer: " << ec.message() << "\n";
+		return;
+	}
+
+	int size = games_and_users_list_.root().children("user").size();
+	std::cout << "player count: " << games_and_users_list_.root().children("user").size() << std::endl;
+	if(size % 2 == 0) {
+		simple_wml::node* dummy_user = games_and_users_list_.root().children("user").at(size-1);
```

Comment:
Would be even nicer if this would cycle through users by removing from the start and adding "player<counter++>" to the end. So it'd start with removing "player0" and then add "player500", remove "player1" and add "player501" etc. That way bugs with issuing commands to the wrong user (like #5840) for example are easy to spot.

Just if you like this is fine as is, too.

Reply:
There's some strange issue going on where
```
dummy_player_count_max_++;
simple_wml::node& dummy_user = games_and_users_list_.root().add_child_at("user", size-1);
dummy_user.set_attr_dup("available", "yes");
dummy_user.set_attr_int("forum_id", dummy_player_count_max_);
dummy_user.set_attr_int("game_id", 0);
dummy_user.set_attr_dup("location", "");
dummy_user.set_attr_dup("moderator", "no");
dummy_user.set_attr_dup("name", ("player"+std::to_string(dummy_player_count_max_)).c_str());
dummy_user.set_attr_dup("registered", "yes");
dummy_user.set_attr_dup("status", "lobby");
```
is somehow duplicating the player name that I myself used to log in (in this case p2), rather than being player501, player502, etc.  So I'd rather leave this for whenever a future change is done to also enable that testing in the first place, since as-is just this wouldn't be enough to allow that testing.